compiler/src/org.graalvm.compiler.nodes.test/src/org/graalvm/compiler/nodes/test/IfNodeCanonicalizationTest2.java

```diff
@@ -0,0 +1,182 @@
+    private static final Class<? extends Node> CLASS = ConditionalNode.class;
+
+    /**
+     * Math.min() tests for integer type, which can be simplified to conditional move.
+     */
+    public int minInt(int m, int n) {
+        return Math.min(m, n);
+    }
+
+    @Test
+    public void testMinInt() {
+        test("minInt", 1, 2);
+        checkNode("minInt", CLASS, 1);
+    }
+
+    public long minLong(long m, long n) {
+        return Math.min(m, n);
+    }
+
+    @Test
+    public void testMinLong() {
+        test("minLong", 1L, 2L);
+        checkNode("minLong", CLASS, 1);
+    }
+
+    /**
+     * Math.max() tests for integer type, which can be simplified to conditional move.
+     */
+    public int maxInt(int m, int n) {
+        return Math.max(m, n);
+    }
+
+    @Test
+    public void testMaxInt() {
+        test("maxInt", 1, 2);
+        checkNode("maxInt", CLASS, 1);
+    }
+
+    public long maxLong(long m, long n) {
+        return Math.max(m, n);
+    }
+
+    @Test
+    public void testMaxLong() {
+        test("maxLong", 1L, 2L);
+        checkNode("maxLong", CLASS, 1);
+    }
+
+    /**
+     * General min test for integer type, which can be simplified to conditional move.
+     */
+    public int generalMin(int m, int n) {
+        int value;
+        if (m <= n) {
+            value = m;
+        } else {
+            value = n;
+        }
+        return 2 * value;
+    }
+
+    @Test
+    public void testGeneralMin() {
+        test("generalMin", 1, 2);
+        checkNode("generalMin", CLASS, 1);
+    }
+
+    /**
+     * General max test for integer type, which can be simplified to conditional move.
+     */
+    public int generalMax(int m, int n) {
+        if (m >= n) {
+            return m;
+        }
+        return n;
+    }
+
+    @Test
+    public void testGeneralMax() {
+        test("generalMax", 1, 2);
+        checkNode("generalMax", CLASS, 1);
+    }
+
+    /**
+     * General conditional set tests for integer type, which can be simplified to conditional move.
+     */
+    public int integerEqualsCondMove(int m, int n, int a, int b) {
+        if (m == n) {
+            return a;
+        }
+        return b;
+    }
+
+    @Test
+    public void testIntegerEqualsCondMove() {
+        test("integerEqualsCondMove", 1, 2, 2, 4);
+        checkNode("integerEqualsCondMove", CLASS, 1);
+    }
+
+    public int isNullCondMove(Object obj, int m, int n) {
+        if (obj == null) {
+            return m;
+        }
+        return n;
+    }
+
+    @Test
+    public void testIsNullCondMove() {
+        test("isNullCondMove", new Object(), 1, 2);
+        checkNode("isNullCondMove", CLASS, 1);
+    }
+
+    /**
+     * IfNode with computations in true and false blocks can't be simplified to ConditionalNode.
+     */
+    public int complexBlock(int m, int n) {
+        if (m <= n) {
+            return m + n;
+        }
+        return m - n;
+    }
+
+    @Test
```

Comment:
Can we remove this test and testMultiPhi since they are negative tests?  We mainly want to test that patterns we are about are optimized.